ludwig/models/trainer.py

```diff
@@ -331,6 +331,60 @@ def write_step_summary(
 
         train_summary_writer.flush()
 
+    def tune_batch_size(
+        self,
+        model,
+        training_set,
+        max_trials: int = 10,
+    ):
+        original_epochs = self.epochs
+        skip_save_model = self.skip_save_model
+        skip_save_progress = self.skip_save_progress
+        skip_save_log = self.skip_save_log
+        # Set temporary values
+        self.epochs = 3
+        self.skip_save_model = True
+        self.skip_save_progress = True
+        self.skip_save_log = True
+
+        high = None
+        count = 0
+        while True:
+            print(count)
+            gc.collect()
+            try:
+                self.train(model, training_set)
+                count += 1
+                if count > max_trials:
+                    break
+                low = self.batch_size
+                prev_batch_size = self.batch_size
+                if high:
+                    if high - low <= 1:
+                        break
+                    midval = (high + low) // 2
+                    self.batch_size = midval
+                else:
+                    self.batch_size *= 2  # double batch size
+
+                if self.batch_size == prev_batch_size:
+                    break
+
+            except tf.errors.ResourceExhaustedError as e:
+                gc.collect()
+                high = prev_batch_size
+                midval = (high + low) // 2
+                self.batch_size = midval
+                if high - low <= 1:
+                    break
+
+        # Restore original parameters to defaults
```

Comment:
We can put this in a `try-finally` block to ensure we restore the params even on exception.

Reply:
Not sure if we need a finally clause here since we restore the original parameters outside of the while loop.